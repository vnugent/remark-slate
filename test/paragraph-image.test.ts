import unified from 'unified';
import markdown from 'remark-parse';
import slate from '../src';

const md = `This is a paragraph.

![my cat one](https://images.unsplash.com/photo-1489084917528-a57e68a79a1e)
![my cat two](https://images.unsplash.com/photo-1514888286974-6c03e2ca1dba)

The end.
`;
it('Use unified to turn markdown into slate state', () => {
  unified()
    .use(markdown)
    .use(slate)
    .process(md, (err, file) => {
      if (err) throw err;
      // @ts-expect-error
      expect(file.result).toMatchSnapshot();
    });
});
