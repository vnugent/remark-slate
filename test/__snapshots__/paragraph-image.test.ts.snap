// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use unified to turn markdown into slate state 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "text": "This is a paragraph.",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "caption": Array [
          Object {
            "text": "my cat one",
          },
        ],
        "children": Array [
          Object {
            "text": "",
          },
        ],
        "link": "https://images.unsplash.com/photo-1489084917528-a57e68a79a1e",
        "type": "image",
      },
      Object {
        "text": "
",
      },
      Object {
        "caption": Array [
          Object {
            "text": "my cat two",
          },
        ],
        "children": Array [
          Object {
            "text": "",
          },
        ],
        "link": "https://images.unsplash.com/photo-1514888286974-6c03e2ca1dba",
        "type": "image",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "The end.",
      },
    ],
    "type": "paragraph",
  },
]
`;
